### PR TITLE
Complete idx assignment

### PR DESCRIPTION
**Pull request by tyb0807**
We also compare the posting list as a way to break ties for `PostingListWithStoppingPoints`. This makes choosing the next top of heap element deterministic (when processing a given stopping point)

### CODE REVIEW

File: rs/index/src/ivf/builder.rs [hunk 1]
```diff
@@ -603,7 +590,27 @@ impl IvfBuilder {
             cur_idx += 1;
         }
 
-        Ok(())
+        Ok(cur_idx)
+    }
+
+    /// Assign new ids to the vectors
+    fn get_reassigned_ids(&mut self) -> Result<Vec<i32>> {
```
Comment (hicder) [inline rs/index/src/ivf/builder.rs:597]: function `get_reassigned_ids` shouldn't do the `assigning`. Let's require explicitly calling `reindex` call first

Comment (hicder) [reply rs/index/src/ivf/builder.rs:597]: the interface should be similar to hnsw

**Review by hicder — CHANGES_REQUESTED** [review]

File: rs/index/src/ivf/builder.rs (hunk 1 above)
Comment (tyb0807) [reply rs/index/src/ivf/builder.rs:597]: I mean this is just to assign the IDs. The reindexing (rearranging vectors in vector storage + updating doc_id_mapping) will be done in `reindex` indeed.

This is the current structure of HNSW.

File: rs/index/src/ivf/builder.rs [hunk 2]
```diff
@@ -515,36 +532,7 @@ impl IvfBuilder {
...
-    //
-    // cur_idx = 9
-    // 20 -> 9 + 5 + 1 = 15
-    // 0 -> 10
-    // 2 -> 11
-    // 4 -> 12
-    // 6 -> 13
-    // 8 -> 14
-    fn reassign_duplicated_vectors(&mut self, assigned_ids: &mut Vec<i32>) -> Result<()> {
+    fn reassign_duplicated_vectors(&mut self, assigned_ids: &mut Vec<i32>) -> Result<i32> {
```
Comment (hicder) [inline rs/index/src/ivf/builder.rs:535]: name this function `assign_ids_until_last_stopping_point`

**Review by hicder — APPROVED** [review]
LGTM

**Pull request merged**